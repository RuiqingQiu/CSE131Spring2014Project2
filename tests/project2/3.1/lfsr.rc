const int c = 50;

function : void printArray(int [c] &a){
   int row = 0;
   int i = -1; while (++i < c){
      cout << a[i] << " ";
      if (++row >= 10){
      cout << endl;
      row = 0;
      }
   }
   cout << endl;
}

function : int lfsr(){
   static int n = 0x1234;
   n = n/2 ^ (-(n & 1) & 0xB400);
   return n;
}

function : int main(){
   int [c] a;
   int i = -1; while (++i < c){
      a[i] = lfsr();
   }
   printArray(a);

   return 0;
}
