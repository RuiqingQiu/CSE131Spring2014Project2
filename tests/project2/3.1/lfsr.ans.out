2330 1165 46662 23331 39313 63688 31844 15922 7961 48012 
24006 12003 41841 58808 29404 14702 7351 47707 59693 49302 
24651 33829 62994 31497 35204 17602 8801 42288 21144 10572 
5286 2643 45353 60564 30282 15141 43410 21705 40548 20274 
10137 42956 21478 10739 41209 58492 29246 14623 43151 57415 

